Tweet Processing Enhancements

jfrank-summit opened the pull request:
This PR addresses two cases where not enough info regarding a tweet was being passed to the LLM:
- Content of quote tweets were not being included
- Excess text in long posts was being ignored

Enhancements to tweet processing:

* [`auto-agents-framework/src/agents/workflows/kol/nodes/engagementNode.ts`](diffhunk://#diff-2b6657036a29ea9e83af31f8d759fbdb53a14a8c6a792cdc252b90264ea50929R15-R24): Added handling for quoted tweets in the `getEngagementDecision` and `createEngagementNode` functions. [[1]](diffhunk://#diff-2b6657036a29ea9e83af31f8d759fbdb53a14a8c6a792cdc252b90264ea50929R15-R24) [[2]](diffhunk://#diff-2b6657036a29ea9e83af31f8d759fbdb53a14a8c6a792cdc252b90264ea50929L53-R76)
* [`auto-agents-framework/src/agents/workflows/kol/nodes/generateTweetNode.ts`](diffhunk://#diff-00ebb8f71961a6907e30b3c7cbcb570e64a57179e95c892ae844a77bf40daaeaR60-R62): Included quoted tweet information in the `postTweet` function.

Improvements to data extraction functions:

* [`auto-agents-framework/src/services/twitter/convertFromTimeline.ts`](diffhunk://#diff-12dde03e4531208a15c78d8aa398ca3078c1df98d6f2f0319c7d552b183694f1L12-R44): Enhanced the `extractLegacyData`, `extractFullText`, and `extractQuotedTweet` functions to handle quoted tweets and prevent infinite recursion.
* [`auto-agents-framework/src/services/twitter/convertFromTimeline.ts`](diffhunk://#diff-12dde03e4531208a15c78d8aa398ca3078c1df98d6f2f0319c7d552b183694f1L26-R53): Updated the `extractPhotos` and `extractVideos` functions to handle cases where media is not available.

Improvements to tweet conversion process:

* [`auto-agents-framework/src/services/twitter/convertFromTimeline.ts`](diffhunk://#diff-12dde03e4531208a15c78d8aa398ca3078c1df98d6f2f0319c7d552b183694f1R71-R110): Modified the `convertTimelineTweetToTweet` function to include additional tweet metadata and handle quoted tweets.

Xm0onh reviewed (commented):
Since we are having new field, could you add types as well?
Like here:
https://github.com/autonomys/autonomys-agents/blob/441164fb2c0451fe6060a3154ce1740a249d03cd/auto-agents-framework/src/agents/workflows/kol/schemas.ts#L8

jfrank-summit commented:
> Since we are having new field, could you add types as well? Like here:
> 
> https://github.com/autonomys/autonomys-agents/blob/441164fb2c0451fe6060a3154ce1740a249d03cd/auto-agents-framework/src/agents/workflows/kol/schemas.ts#L8

I have added a type for quotedTweet and fixed an issue with failed timeline tweet conversions.

Xm0onh commented on auto-agents-framework/src/services/twitter/convertFromTimeline.ts:
```diff
@@ -41,30 +65,45 @@ const extractUrls = (entities: any) =>
 
   return {
-    id: tweet.rest_id,
+    id: tweet.rest_id || legacy.id_str,
```
Does Twitter send back different structure? what was the problem before, because I didn't have any issue with `id` field.

jfrank-summit replied:
Just extra redundancy, I don't trust the timeline tweet structure.

Xm0onh reviewed (dismissed):
LGTM.
Left some questions for my understanding.

jfrank-summit commented on auto-agents-framework/src/services/twitter/client.ts:
```diff
@@ -257,12 +259,18 @@ export const createTwitterApi = async (
+      return tweets
+        .filter(isValidTweet)
+        .map(tweet => convertTimelineTweetToTweet(tweet))
+        .filter(tweet => tweet !== undefined);
```
There was an uncovered case that was causing failures. Digging into it I was able to add processing of more timeline tweets rather than fail if they don't have the `__typename` `Tweet`.
